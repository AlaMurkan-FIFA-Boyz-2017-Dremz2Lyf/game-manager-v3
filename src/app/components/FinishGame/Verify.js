import React from 'react';
import PropTypes from 'prop-types';
import { Header, Segment, Table } from 'semantic-ui-react';

const TRACKED_STATS = [
  'score',
  'shots',
  'shots on target',
  'possession %',
  'tackles',
  'fouls',
  'yellow cards',
  'red cards',
  'injuries',
  'offsides',
  'corners',
  'shot accuracy %',
  'pass accuracy %',
];

const Verify = ({ results }) => (
  <Segment basic>
    <Header color="teal">
      Verify
    </Header>
    <p>Please verify that the results match the image below.</p>
    <Table compact inverted unstackable>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell textAlign="left">Home</Table.HeaderCell>
          <Table.HeaderCell textAlign="center">Stat</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">Away</Table.HeaderCell>
        </Table.Row>
      </Table.Header>

      <Table.Body>
        {TRACKED_STATS.map((statName) => {
          const home = results.home[statName];
          const away = results.away[statName];
          return (
            <Table.Row key={statName}>
              <Table.Cell textAlign="left">{home}</Table.Cell>
              <Table.Cell textAlign="center">{statName}</Table.Cell>
              <Table.Cell textAlign="right">{away}</Table.Cell>
            </Table.Row>
          );
        })}
      </Table.Body>
    </Table>
  </Segment>
);

Verify.propTypes = {
  results: PropTypes.object,
};

export default Verify;
