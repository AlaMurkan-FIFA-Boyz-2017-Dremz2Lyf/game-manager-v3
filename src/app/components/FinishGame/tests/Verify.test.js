import React from 'react';
import { shallow } from 'enzyme';

import Verify from '../Verify';
import { combinedResults } from './tesseractMock';

describe('Verify', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Verify results={combinedResults} />,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('TableRow').length).toBe(14);
  });
});
