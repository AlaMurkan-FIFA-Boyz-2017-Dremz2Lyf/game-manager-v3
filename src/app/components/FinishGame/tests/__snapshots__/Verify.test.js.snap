// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify should match the snapshot 1`] = `
<Segment
  basic={true}
>
  <Header
    color="teal"
  >
    Verify
  </Header>
  <p>
    Please verify that the results match the image below.
  </p>
  <Table
    as="table"
    compact={true}
    inverted={true}
    unstackable={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
          textAlign="left"
        >
          Home
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          textAlign="center"
        >
          Stat
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          textAlign="right"
        >
          Away
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="score"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          0
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          score
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          4
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="shots"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          7
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          shots
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          12
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="shots on target"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          1
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          shots on target
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          7
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="possession %"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          37
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          possession %
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          63
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="tackles"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          7
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          tackles
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          2
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="fouls"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          7
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          fouls
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          1
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="yellow cards"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          1
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          yellow cards
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          0
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="red cards"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          0
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          red cards
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          0
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="injuries"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          0
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          injuries
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          1
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="offsides"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          0
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          offsides
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          0
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="corners"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          0
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          corners
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          2
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="shot accuracy %"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          14
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          shot accuracy %
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          58
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="pass accuracy %"
      >
        <TableCell
          as="td"
          textAlign="left"
        >
          77
        </TableCell>
        <TableCell
          as="td"
          textAlign="center"
        >
          pass accuracy %
        </TableCell>
        <TableCell
          as="td"
          textAlign="right"
        >
          84
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Segment>
`;
